janitor/functions/mutate.py

```diff
@@ -0,0 +1,167 @@
+"""Function for mutation of a column or columns."""
+from typing import Any
+import pandas as pd
+import pandas_flavor as pf
+
+from janitor.utils import check
+
+from janitor.functions.utils import SD, _process_SD
+from itertools import product
+
+
+@pf.register_dataframe_method
+def mutate(
+    df: pd.DataFrame,
+    *args,
+    by: Any = None,
+) -> pd.DataFrame:
+    """
+
+    !!! info "New in version 0.25.0"
+
+    !!!note
+
+        Before reaching for `mutate`, try `pd.DataFrame.assign`.
+
+    Transform columns via a tuple.
+
+    The computation should return a 1-D array like object
+    that is the same length as `df` or a scalar
+    that can be broadcasted to the same length as `df`.
+
+    The argument should be of the form `(columns, func, names_glue)`;
```

Comment:
```suggestion
    The argument should be of the form `(columns, mutation_func, names_glue)`;
```

I changed the name here b/c I thought it would be clearer. However, this is mostly cosmetic (IMO), and it does break the consistency in how we use `func` to denote a function, so it's okay to not accept this suggestion.